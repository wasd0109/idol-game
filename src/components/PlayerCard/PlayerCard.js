import React from 'react';

import profileImage from '../../assets/profile_image.jpg';

function PlayerCard({ player }) {
  const displayInfo = ['name', 'title', 'level', 'element', 'message'];
  const stats = [];
  for (const [key, value] of Object.entries(player)) {
    if (displayInfo.includes(key)) stats.push([key.replace('_', ' '), value]);
  }
  return (
    <div className="my-2 w-8/12 h-auto border-l-2 border-t-2 m-auto flex">
      <div className="justify-start w-2/12 h-auto border-r-2 border-b-2">
        <img src={profileImage} alt="Profile" />
      </div>
      <div className="grid w-10/12 grid-rows-3  grid-cols-2">
        {stats.map((stat, i) => (
          <p
            key={stat[0] + i}
            className={`row-span-1 ${
              stat[0] === 'message' ? 'col-span-2' : 'col-span-1 '
            } capitalize border-r-2 border-b-2`}
          >
            {stat[0]}: <span className="text-lg font-bold">{stat[1]}</span>
          </p>
        ))}
      </div>
    </div>
  );
}

export default PlayerCard;
