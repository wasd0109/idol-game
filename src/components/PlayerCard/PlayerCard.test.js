import React from 'react';
import { render, screen } from '@testing-library/react';
import PlayerCard from './PlayerCard';

const initialProps = {
  player: {
    id: 0,
    name: 'Ken',
    title: 'rookie',
    level: '47',
    element: 'fire',
    message: 'hello',
  },
};

describe('PlayerCard component render correctly', () => {
  test('PlayerCard component render', () => {
    expect(render(<PlayerCard {...initialProps} />)).toMatchSnapshot();
  });
});
