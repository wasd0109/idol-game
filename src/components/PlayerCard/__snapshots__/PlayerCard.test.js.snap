// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCard component render correctly PlayerCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-2 w-8/12 h-auto border-l-2 border-t-2 m-auto flex"
      >
        <div
          class="justify-start w-2/12 h-auto border-r-2 border-b-2"
        >
          <img
            alt="Profile"
            src="profile_image.jpg"
          />
        </div>
        <div
          class="grid w-10/12 grid-rows-3  grid-cols-2"
        >
          <p
            class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
          >
            name
            : 
            <span
              class="text-lg font-bold"
            >
              Ken
            </span>
          </p>
          <p
            class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
          >
            title
            : 
            <span
              class="text-lg font-bold"
            >
              rookie
            </span>
          </p>
          <p
            class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
          >
            level
            : 
            <span
              class="text-lg font-bold"
            >
              47
            </span>
          </p>
          <p
            class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
          >
            element
            : 
            <span
              class="text-lg font-bold"
            >
              fire
            </span>
          </p>
          <p
            class="row-span-1 col-span-2 capitalize border-r-2 border-b-2"
          >
            message
            : 
            <span
              class="text-lg font-bold"
            >
              hello
            </span>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-2 w-8/12 h-auto border-l-2 border-t-2 m-auto flex"
    >
      <div
        class="justify-start w-2/12 h-auto border-r-2 border-b-2"
      >
        <img
          alt="Profile"
          src="profile_image.jpg"
        />
      </div>
      <div
        class="grid w-10/12 grid-rows-3  grid-cols-2"
      >
        <p
          class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
        >
          name
          : 
          <span
            class="text-lg font-bold"
          >
            Ken
          </span>
        </p>
        <p
          class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
        >
          title
          : 
          <span
            class="text-lg font-bold"
          >
            rookie
          </span>
        </p>
        <p
          class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
        >
          level
          : 
          <span
            class="text-lg font-bold"
          >
            47
          </span>
        </p>
        <p
          class="row-span-1 col-span-1  capitalize border-r-2 border-b-2"
        >
          element
          : 
          <span
            class="text-lg font-bold"
          >
            fire
          </span>
        </p>
        <p
          class="row-span-1 col-span-2 capitalize border-r-2 border-b-2"
        >
          message
          : 
          <span
            class="text-lg font-bold"
          >
            hello
          </span>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
